### PR TITLE
fix: Builder

### PR DESCRIPTION
**Pull request by bitsandfoxes**
#skip-changelog

### CODE REVIEW

File: test/Sentry.Unity.Tests/IntegrationTests.cs [hunk 1]
```diff
@@ -274,6 +275,8 @@ public IEnumerator DebugLogException_OnMainThread_IsCapturedAndIsMainThreadIsTru
     [UnityTest]
     public IEnumerator DebugLogException_InTask_IsCapturedAndIsMainThreadIsFalse()
     {
+        LogAssert.Expect(LogType.Exception, new Regex("Exception: .* Test Event"));
```
Comment (bitsandfoxes) [inline test/Sentry.Unity.Tests/IntegrationTests.cs:278]: Fixed a flaky test while at it.

**Review by bruno-garcia — APPROVED** [review]
This removes the integration tests on the built UPM package, right? please add a ticket to bring it back so we don't forget

**Comment by bitsandfoxes** [comment]
> This removes the integration tests on the built UPM package, right? please add a ticket to bring it back so we don't forget 

The tests stay the same. I removed the duplicated logic of `is/is not main` between the `create project` and the `build project` steps.

File: test/Sentry.Unity.Tests/IntegrationTests.cs (hunk 1 above)
Comment (bruno-garcia) [reply test/Sentry.Unity.Tests/IntegrationTests.cs:278]: Not sure this fixed it: 
image
https://github.com/getsentry/sentry-unity/actions/runs/11224732057/job/31202068887

**Pull request merged**